azurerm/internal/services/securitycenter/security_center_contact_resource_test.go

```diff
@@ -1,188 +1,130 @@
-					resource.TestCheckResourceAttr(data.ResourceName, "email", "basic@example.com"),
-					resource.TestCheckResourceAttr(data.ResourceName, "phone", "+1-555-555-5555"),
-					resource.TestCheckResourceAttr(data.ResourceName, "alert_notifications", "true"),
-					resource.TestCheckResourceAttr(data.ResourceName, "alerts_to_admins", "true"),
-				),
-			},
-			data.ImportStep(),
+	r := SecurityCenterContactResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.template("basic@example.com", "+1-555-555-5555", true, true),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("email").HasValue("basic@example.com"),
+				check.That(data.ResourceName).Key("phone").HasValue("+1-555-555-5555"),
+				check.That(data.ResourceName).Key("alert_notifications").HasValue("true"),
+				check.That(data.ResourceName).Key("alerts_to_admins").HasValue("true"),
+			),
 		},
+		data.ImportStep(),
 	})
 }
 
-func testAccAzureRMSecurityCenterContact_requiresImport(t *testing.T) {
+func testAccSecurityCenterContact_requiresImport(t *testing.T) {
 	data := acceptance.BuildTestData(t, "azurerm_security_center_contact", "test")
-	resource.Test(t, resource.TestCase{
-		PreCheck:     func() { acceptance.PreCheck(t) },
-		Providers:    acceptance.SupportedProviders,
-		CheckDestroy: testCheckAzureRMSecurityCenterContactDestroy,
-		Steps: []resource.TestStep{
-			{
-				Config: testAccAzureRMSecurityCenterContact_template("require@example.com", "+1-555-555-5555", true, true),
-				Check: resource.ComposeTestCheckFunc(
-					testCheckAzureRMSecurityCenterContactExists(data.ResourceName),
-					resource.TestCheckResourceAttr(data.ResourceName, "email", "require@example.com"),
-					resource.TestCheckResourceAttr(data.ResourceName, "phone", "+1-555-555-5555"),
-					resource.TestCheckResourceAttr(data.ResourceName, "alert_notifications", "true"),
-					resource.TestCheckResourceAttr(data.ResourceName, "alerts_to_admins", "true"),
-				),
-			},
-			data.RequiresImportErrorStep(func(data acceptance.TestData) string {
-				return testAccAzureRMSecurityCenterContact_requiresImportCfg("email1@example.com", "+1-555-555-5555", true, true)
-			}),
+	r := SecurityCenterContactResource{}
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.template("require@example.com", "+1-555-555-5555", true, true),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("email").HasValue("require@example.com"),
+				check.That(data.ResourceName).Key("phone").HasValue("+1-555-555-5555"),
+				check.That(data.ResourceName).Key("alert_notifications").HasValue("true"),
+				check.That(data.ResourceName).Key("alerts_to_admins").HasValue("true"),
+			),
 		},
+		data.RequiresImportErrorStepWithConfig(r.requiresImportCfg("email1@example.com", "+1-555-555-5555", true, true)),
```

Comment:
rather than doing this, we can make this:

```suggestion
		data.RequiresImportStep(func(data acceptance.TestData) string {
			return r.requiresImportCfg("email1@example.com", "+1-555-555-5555", true, true)
		}),
```

which means we can remove this overload, since everything should be using TestData here